cognite/client/_api/data_modeling/streams.py

```diff
@@ -0,0 +1,180 @@
+from __future__ import annotations
+
+from collections.abc import Iterator, Sequence
+from typing import TYPE_CHECKING, overload
+
+from cognite.client._api_client import APIClient
+from cognite.client._constants import DEFAULT_LIMIT_READ
+from cognite.client.data_classes.data_modeling.streams import Stream, StreamList, StreamWrite
+from cognite.client.utils._experimental import FeaturePreviewWarning, warn_on_all_method_invocations
+from cognite.client.utils._identifier import Identifier
+
+if TYPE_CHECKING:
+    from cognite.client import CogniteClient
+    from cognite.client.config import ClientConfig
+
+
+@warn_on_all_method_invocations(
+    FeaturePreviewWarning(api_maturity="alpha", sdk_maturity="alpha", feature_name="Records API")
+)
+class StreamsAPI(APIClient):
+    _RESOURCE_PATH = "/streams"
+
+    def __init__(self, config: ClientConfig, api_version: str | None, cognite_client: CogniteClient) -> None:
+        super().__init__(config, api_version, cognite_client)
+        self._CREATE_LIMIT = 1
+        self.__alpha_headers = {
+            "cdf-version": "alpha",
+        }
+
+    @overload
+    def __call__(
+        self,
+        chunk_size: None = None,
+        limit: int | None = None,
+    ) -> Iterator[Stream]: ...
+
+    @overload
+    def __call__(
+        self,
+        chunk_size: int,
+        limit: int | None = None,
+    ) -> Iterator[StreamList]: ...
+
+    def __call__(
+        self,
+        chunk_size: int | None = None,
+        limit: int | None = None,
+    ) -> Iterator[Stream] | Iterator[StreamList]:
+        """Iterate over streams
+
+        Fetches streams as they are iterated over, so you keep a limited number of streams in memory.
+
+        Args:
+            chunk_size (int | None): Number of streams to return in each chunk. Defaults to yielding one stream a time.
+            limit (int | None): Maximum number of streams to return. Defaults to returning all items.
+
+        Returns:
+            Iterator[Stream] | Iterator[StreamList]: yields Stream one by one if chunk_size is not specified, else StreamList objects.
+        """
+        return self._list_generator(
+            list_cls=StreamList,
+            resource_cls=Stream,
+            method="GET",
+            chunk_size=chunk_size,
+            limit=limit,
```

Comment:
note: we do not have limit on listStream atm.

https://api-docs.cogheim.net/redoc/#tag/Streams/operation/listStream